src/qunit.cpp

```diff
@@ -756,36 +756,26 @@ bool QUnit::TrySeparate(bitLenInt qubit)
         }
     }
 
-    real1_f r = sqrt(probZ * probZ + probX * probX + probY * probY);
+    const real1_f r = sqrt(probZ * probZ + probX * probX + probY * probY);
     if ((ONE_R1 / 2 - r) > separabilityThreshold) {
         return false;
     }
 
-    real1_f inclination = atan2(sqrt(probX * probX + probY * probY), probZ);
-    real1_f azimuth = atan2(probY, probX);
-
-    if (std::isnan(inclination) || std::isinf(inclination) || std::isnan(azimuth) || std::isinf(azimuth)) {
-        return false;
-    }
-
     RevertBasis1Qb(qubit);
 
-    shard.unit->IAI(shard.mapped, azimuth, inclination);
-    shard.isProbDirty = true;
+    const real1_f inclination = atan2(sqrt(probX * probX + probY * probY), probZ);
+    const real1_f azimuth = atan2(probY, probX);
 
-    prob = (ONE_R1 / 2) - ProbBase(qubit);
-    if (!shard.unit) {
-        ShardAI(shard, azimuth, inclination);
-        return true;
-    }
+    shard.unit->IAI(shard.mapped, azimuth, inclination);
 
-    bool value = prob < ZERO_R1;
+    prob = (ONE_R1 / 2) - shard.Prob();
```

Comment:
This isn't right. If I check the `unit` probability for the mapped bit index instead of this, which relies on shard cache, the test fails. Still WIP.